web/source/osk/visualKeyboard.ts

```diff
@@ -1789,8 +1789,12 @@ namespace com.keyman.osk {
 
       // And correct its position with respect to that element
       ss=subKeys.style;
-      var x=dom.Utils.getAbsoluteX(e)+0.5*(e.offsetWidth-subKeys.offsetWidth), y,
-        xMax=(util.landscapeView()?screen.height:screen.width)-subKeys.offsetWidth;
+      var x=dom.Utils.getAbsoluteX(e)+0.5*(e.offsetWidth-subKeys.offsetWidth);
+      
+      // https://stackoverflow.com/questions/28083715/how-to-detect-if-a-mobile-device-is-emulated-by-google-chrome
+      // (Note the last reply to the accepted answer.)
```

Comment:
```suggestion
      // Note in particular https://stackoverflow.com/questions/28083715/#comment108747578_28084346:
      // The definitive way to determine this is to check for `navigator.maxTouchPoints > 1` Because Chrome 
      // emulated devices will always have only 1 touchpoint (ie. your mouse cursor), while actual (modern)
      // mobile devices will always have more than 1, and devices without touch support will always have 
      // 0. – Ivan [CC-BY-SA 4.0]
```

And yet I wonder if this is true on a Windows computer with a touch screen.

I will try and test this on my wife's touch-screen laptop shortly.

Reply:
If `false` in that case, it'll still fail during Chrome _emulation_ of an iOS device.  Fortunately, all standard user use-cases will be fine, though it could affect keyboard developers like Makara when using the Developer test host.

Reply:
I have verified that `navigator.maxTouchPoints` is 10 on my wife's computer when mobile emulation is *not* running, but it is 1 when mobile emulation is running. So that's good news.